src/hostnet/lib/host_uwt.ml

```diff
@@ -82,11 +82,14 @@ module Sockets = struct
         loop () in
       loop ()
 
-    let input ~reply ~src:(src, src_port) ~dst:(dst, dst_port) ~payload =
+    let input ?desc ~reply ~src:(src, src_port) ~dst:(dst, dst_port) ~payload () =
       (if Hashtbl.mem table (src, src_port, dst, dst_port) then begin
           Lwt.return (Some (Hashtbl.find table (src, src_port, dst, dst_port)))
         end else begin
-         let description = Ipaddr.to_string src ^ ":" ^ (string_of_int src_port) ^ "-" ^ (Ipaddr.to_string dst) ^ ":" ^ (string_of_int dst_port) in
+         let base_description = Ipaddr.to_string src ^ ":" ^ (string_of_int src_port) ^ "-" ^ (Ipaddr.to_string dst) ^ ":" ^ (string_of_int dst_port) in
```

Comment:
This does a large number of intermediate string allocations on the input path. Would be better to replace any >size 2 concats with:

```
let base_description =
  String.concat [ el1; el2; ...; eln ]
```

but also if it's not used in the main datapath outside of logging, put this allocation inside the call to `Log.debug` (a few lines below) so that it will only get evaluated if the logger is active.

```
Log.debug (fun f -> foo ())
```

will only evaluate `foo ()` if debug logging is active.